WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserAdapter.java

```diff
@@ -61,36 +85,90 @@ public ThemeBrowserAdapter(Context context, Cursor c, boolean autoRequery, Theme
         configureThemeImageSize(parent);
         ThemeViewHolder themeViewHolder = new ThemeViewHolder(view);
         view.setTag(themeViewHolder);
-
         return view;
     }
 
     @Override
     public void bindView(View view, Context context, Cursor cursor) {
-        final ThemeViewHolder themeViewHolder = (ThemeViewHolder) view.getTag();
+        if (getItemViewType(cursor) == HEADER_VIEW_TYPE) {
+            final HeaderViewHolder headerViewHolder = (HeaderViewHolder) view.getTag();
+            final String headerText = cursor.getString(cursor.getColumnIndex(ThemeModelTable.NAME));
+            final String countText = cursor.getString(cursor.getColumnIndex("count"));
+            headerViewHolder.headerText.setText(headerText);
+            headerViewHolder.countText.setText(countText);
+            return;
+        }
 
-        final String screenshotURL = cursor.getString(cursor.getColumnIndex(Theme.SCREENSHOT));
-        final String name = cursor.getString(cursor.getColumnIndex(Theme.NAME));
-        final String price = cursor.getString(cursor.getColumnIndex(Theme.PRICE));
-        final String themeId = cursor.getString(cursor.getColumnIndex(Theme.ID));
-        final boolean isCurrent = cursor.getInt(cursor.getColumnIndex(Theme.IS_CURRENT)) == 1;
-        final boolean isPremium = !price.isEmpty();
+        String screenshotURL = cursor.getString(cursor.getColumnIndex(ThemeModelTable.SCREENSHOT_URL));
+        final ThemeViewHolder themeViewHolder = (ThemeViewHolder) view.getTag();
+        final String name = cursor.getString(cursor.getColumnIndex(ThemeModelTable.NAME));
+        final String themeId = cursor.getString(cursor.getColumnIndex(ThemeModelTable.THEME_ID));
+        final String currency = cursor.getString(cursor.getColumnIndex(ThemeModelTable.CURRENCY));
+        final float price = cursor.getFloat(cursor.getColumnIndex(ThemeModelTable.PRICE));
+        final boolean isPremium = price > 0.f;
+        final boolean isCurrent =
+                StringUtils.equals("true", cursor.getString(cursor.getColumnIndex(ThemeModelTable.ACTIVE)));
 
         themeViewHolder.nameView.setText(name);
-        themeViewHolder.priceView.setText(price);
+        if (isPremium) {
+            String priceText = currency + String.valueOf((int) price);
+            themeViewHolder.priceView.setText(priceText);
+            themeViewHolder.priceView.setVisibility(View.VISIBLE);
+        } else {
+            themeViewHolder.priceView.setVisibility(View.GONE);
+        }
+
+        // catch the case where a URL has no protocol
+        if (!screenshotURL.startsWith("http")) {
+            // some APIs return a URL starting with // so the protocol can be supplied by the client
+            // strip // before adding the protocol
+            if (screenshotURL.startsWith("//")) {
+                screenshotURL = screenshotURL.substring(2);
+            }
+            screenshotURL = "https://" + screenshotURL;
```

Comment:
There is `ThemeWebActivity.THEME_HTTPS_PREFIX` for this.

And can you create one for "http" as well just for cleanliness on line 145 please?